File: javascript/bob/bob.js
var Bob = function(){
  this.hey = function(words){
    if( /^\s*$/.test(words))
      return 'Fine. Be that way!';
    else if(words == words.toUpperCase() && /[A-Z]/.test(words) )
      return 'Woah, chill out!';
    else if(/\?$/.test(words))
      return 'Sure.';
    else 
      return 'Whatever.';
  };
};

module.exports = Bob;

File: elixir/bob/bob.exs

defmodule Teenager do
  @moduledoc """
  Elixir Teenager module 
  For instance; 
      def hey(input) do
      end
  """

  @doc """
  Greets the user by name, or by saying "Hello, World!"
  if no name is given.
  """
  @spec hey(String.t) :: String.t
  def hey(input) do
    cond do
      input =~ ~r/^\s*$/ ->  "Fine. Be that way!"
      input =~ ~r/^[A-Z0-9 %^*@#$(!]+$/ -> "Woah, chill out!"
      input =~ ~r/\?$/ -> "Sure."
      true -> "Whatever."
    end
  end
end
